programs/uxd/src/utils/calculate_depositories_mint_collateral_amount.rs

```diff
@@ -0,0 +1,127 @@
+use anchor_lang::prelude::Result;
+use anchor_lang::require;
+
+use crate::error::UxdError;
+use crate::utils::calculate_depositories_sum_value;
+
+use super::checked_convert_u128_to_u64;
+use super::compute_amount_less_fraction_floor;
+
+pub struct DepositoriesMintCollateralAmount {
+    pub identity_depository_mint_collateral_amount: u64,
+    pub mercurial_vault_depository_0_mint_collateral_amount: u64,
+    pub credix_lp_depository_0_mint_collateral_amount: u64,
+}
+
+pub fn calculate_depositories_mint_collateral_amount(
+    requested_mint_collateral_amount: u64,
+    identity_depository_target_redeemable_amount: u64,
+    mercurial_vault_depository_0_target_redeemable_amount: u64,
+    credix_lp_depository_0_target_redeemable_amount: u64,
+    identity_depository_redeemable_amount_under_management: u128,
+    mercurial_vault_depository_0_redeemable_amount_under_management: u128,
+    credix_lp_depository_0_redeemable_amount_under_management: u128,
+) -> Result<DepositoriesMintCollateralAmount> {
+    // ---------------------------------------------------------------------
+    // -- Phase 1
+    // -- Calculate the maximum mintable collateral amount for each depository
+    // ---------------------------------------------------------------------
+
+    let identity_depository_mintable_collateral_amount =
+        calculate_depository_mintable_collateral_amount(
+            identity_depository_redeemable_amount_under_management,
+            identity_depository_target_redeemable_amount,
+        )?;
+    let mercurial_vault_depository_0_mintable_collateral_amount =
+        calculate_depository_mintable_collateral_amount(
+            mercurial_vault_depository_0_redeemable_amount_under_management,
+            mercurial_vault_depository_0_target_redeemable_amount,
+        )?;
+    let credix_lp_depository_0_mintable_collateral_amount =
+        calculate_depository_mintable_collateral_amount(
+            credix_lp_depository_0_redeemable_amount_under_management,
+            credix_lp_depository_0_target_redeemable_amount,
+        )?;
+
+    // ---------------------------------------------------------------------
+    // -- Phase 2
+    // -- Calculate the total amount we could possibly mint
+    // -- If this total is not enough, we abort
+    // ---------------------------------------------------------------------
+
+    let total_mintable_collateral_amount = calculate_depositories_sum_value(
+        identity_depository_mintable_collateral_amount,
```

Comment:
So yes here it should error if we add a new depository and forget to edit i believe else it’s hard to keep track?

Reply:
exactly, that was my point, make it compiler-enforced that the algorithm take into account all the necessary depositories